### PR TITLE
Story/gra 1252

### PR DESCRIPTION
**Pull request by afeiszli**
## Describe your changes

This modifies the nm-quick script to do the following:
1. Adds a "-a" flag, which will run "auto install" with default options. No prompts will be used.
2. Installs the netclient into the docker-compose, makes it a default host and an ingress gw

The nm-quick script was modified to use functions, rather than one long script, which is why the changes look so big, but it is mostly just reformatting the existing script.

This also makes a small change to nm-upgrade to replace some faulty logic, and adds an example netclient docker-compose file to the compose folder.

## Provide Issue ticket number if applicable/not in title

GRA-1252 (this is one part of a bigger ticket)

## Provide testing steps

1. Run "./nm-quick.sh -a" on a fresh linux machine
2. See that netmaker successfully installs, and there is a healthy netclient running

## Checklist before requesting a review
- [x] My changes affect only 10 files or less.
- [x] I have performed a self-review of my code and tested it.
- [ ] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [ ] If it is a bugfix, my code is <= 200 lines.
- [ ] My functions are <= 80 lines.
- [ ] I have had my code reviewed by a peer.
- [ ] My unit tests pass locally.
- [x] Netmaker is awesome.

### CODE REVIEW

**Review by mattkasun — COMMENTED** [review]
how is netclient going to work in a container?
the netclient.sh script tries to join using a token and if it fails it exits

**Comment by theguy951357** [comment]
will wait on changes.

**Comment by theguy951357** [comment]
I did test this one. the script runs well. the netmaker-1 node is connected.
I can ping the netmaker-1 and the netmaker-1 can ping other peers.
ingress and egress are not reachable when created in netmaker-1.
netmaker-1 can reach an ingress and egress that are created on other peers.

can test again after comments addressed and merge conflict fixed

**Comment by afeiszli** [comment]
The iptables commands in docker netclient are not working properly with the host. I've moved it back to a host-level install for now, which appears to be working fine. Will re-attempt a docker installation in next release once we've sorted out issues with iptables in docker.

**Comment by afeiszli** [comment]
@theguy951357 can we get a retest on this please?

**Comment by theguy951357** [comment]
ok testing now.

**Review by theguy951357 — APPROVED** [review]
netmaker server is up and running
netclient is up and running
pr checklist 100%

**Comment by afeiszli** [comment]
@mattkasun removed container portion

File: compose/docker-compose.netclient.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+version: "3.4"
+
+services:
+  netclient:
+    container_name: netclient
+    image: 'gravitl/netclient:v0.18.1'
+    hostname: netmaker-1
+    network_mode: host
+    depends_on:
```
Comment (mattkasun) [inline compose/docker-compose.netclient.yml:9]: why would this depend on netmaker?

**Review by mattkasun — CHANGES_REQUESTED** [review]

File: compose/docker-compose.netclient.yml (hunk 1 above)
Comment (afeiszli) [reply compose/docker-compose.netclient.yml]: fixed

**Review by mattkasun — APPROVED** [review]

**Pull request merged**